CHANGELOG.md

```diff
@@ -1,6 +1,47 @@
 # go-ipfs changelog
 
-### 0.4.10 - 2017-06-18
+### 0.4.10 - 2017-06-27
+
+Ipfs 0.4.10 is a patch release that contains several exciting new features,
+bugfixes and general improvements.
+
+The `ipfs pin` command got two new subcommands, `verify` and `update`. `ipfs
+pin verify` is used to scan the repo for pinned object graphs and check their
+integrity. Any issues are reported back in a way that should make it relatively
+easy to recover from. This subcommand was added to help recover from datastore
+corruptions, particularly if using the experimental filestore and accidentally
+deleting tracked files.
+`ipfs pin update` was added to make the task of keeping a large, frequently
+changing object graph pinned. Previously users had to call `ipfs pin rm` on the
+old pin, and `ipfs pin add` on the new one. The 'new' `ipfs pin add` call would
+be very expensive as it would need to verify the entirety of the graph again.
+The `ipfs pin update` command takes shortcuts, portions of the graph that were
+covered under the old pin are assumed to be fine, and the command skips
+checking them.
+
+Next up, we have finally implemented an `ipfs shutdown` command so users can
+shut down their ipfs daemons via the api. This is especially useful on
+platforms that make it difficult to control processes (android, for example),
+and is also useful when needing to shut down a node remotely and you do not
+have access to the machine itself (just the api).
+
+`ipfs add` has gained a new flag, the `--hash` flag allows you to select which
```

Comment:
gonna use a semicolon instead of a period. The sentence feels incomplete just saying "it got a new flag" without describing what that flag is.